### DISCUSSION TITLE
Is there any way to update the `validate_user` method?

### DISCUSSION THREAD

**Discussion by rohit-uf** [oidc-wp/openid-connect-generic#437] [Q&A] [answered]
## Description

In the `OpenID_Connect_Generic_Client_Wrapper` class, there is a method called [validate_user($user)](https://github.com/oidc-wp/openid-connect-generic/blob/dev/includes/openid-connect-generic-client-wrapper.php#L586-L593).

## Issue/Enhancement
Currently it is only checking if the passed object is of type User and that the user exists. But how can I customize this logic

For example, if the users are coming from another website, and only those users who have performed an action on that website should be allowed to login on the current website.  How can I achieve this?

**Comment by timnolte** [comment, marked as answer]
@rohit-uf I moved your question to the Q&A Discussion area as this is not an Issue.

To answer your question there are many Hooks present in the plugin, some being there to handle special logic like you are asking.

https://github.com/oidc-wp/openid-connect-generic/wiki/Hooks---Actions-&-Filters

**Reply by rohit-uf** [reply]
https://github.com/oidc-wp/openid-connect-generic/wiki/Hooks---Actions-&-Filters#openid-connect-generic-user-login-test

The filter `openid-connect-generic-user-login-test` was exactly what I needed. Sorry for missing it earlier 😅 .

Thanks a lot
